### ISSUE TITLE
Remove Replace=true in favor of Server-Side Apply syncs

### ISSUE THREAD

**Issue by leoluz** [argoproj/argo-cd#11573]
# Summary

Remove the `Replace=true` sync option from Argo CD.

# Motivation

The sync option to replace the resource state leads to several issues:
- https://github.com/argoproj/argo-cd/issues/9913
- https://github.com/argoproj/argo-cd/issues/7131
- https://github.com/argoproj/argo-cd/issues/9715

We recently had an production outage caused by a sync mistakenly executed with `Replace=true`.

# Proposal

First deprecate the `Replace=true` sync option.
Stabilize Server-Side Apply addressing existing bugs.
Remove `Replace=true` when SSA is stable.

**Comment by n9** [comment]
@leoluz After reading the tickets you have linked. It seems to me that the issue is caused by using client-side apply.

https://github.com/argoproj/argo-cd/issues/9913 - issue when using client-side apply after replace
https://github.com/argoproj/argo-cd/issues/7131- the issue was cause because the `replace=true; was not working (ArgoCD was still using client-side apply https://github.com/argoproj/argo-cd/pull/7137)
https://github.com/argoproj/argo-cd/issues/9715 - the issue suggest to always use `replace=true` for CRDs

Or am I wrong?

**Comment by leoluz** [comment]
@n9 Correct. My point was that we need to stabilize/improve server-side apply feature in Argo CD. Basically the diff calculation for SSA needs a different strategy imo. See https://github.com/argoproj/argo-cd/issues/11574
